auditbeat/module/auditd/audit_linux.go

```diff
@@ -154,21 +154,27 @@ func closeAuditClient(client *libaudit.AuditClient, log *logp.Logger) {
 	// Drain the netlink channel in parallel to Close() to prevent a deadlock.
 	// This goroutine will terminate once receive from netlink errors (EBADF,
 	// EBADFD, or any other error). This happens because the fd is closed.
-	go func() {
+	closeWaiter := &sync.WaitGroup{}
+	closeWaiter.Add(1)
+	go func(testClient *libaudit.AuditClient) {
 		for {
-			_, err := client.Netlink.Receive(true, discard)
+			_, err := testClient.Netlink.Receive(true, discard)
 			switch {
 			case err == nil, errors.Is(err, syscall.EINTR):
 			case errors.Is(err, syscall.EAGAIN):
 				time.Sleep(50 * time.Millisecond)
 			default:
+				closeWaiter.Done()
 				return
 			}
 		}
-	}()
+	}(client)
 	if err := client.Close(); err != nil {
 		log.Errorw("Error closing audit monitoring client", "error", err)
 	}
+	// If we don't wait for the socket to drain, the calling function may tinker with
+	// the AuditClient pointer while the goroutine is trying to drain it
```

Comment:
Can you clarify this comment? AFAICS the `*libaudit.AuditClient` that's passed in here is never altered after the call returns.